rust/operator-binary/src/controller.rs

```diff
@@ -358,3 +430,12 @@ fn build_opa_start_command() -> Vec<String> {
         "/stackable/config/config.yaml".to_string(),
     ]
 }
+
+fn service_ports() -> Vec<ServicePort> {
+    vec![ServicePort {
```

Comment:
Don't you need to add the metrics port here too?

Reply:
It is not possible. You cannot have the same port with different names. It should be no problem though since for the prometheus Pod/ServiceMonitor you can define the port name yourself.

Reply:
> It should be no problem though since for the prometheus Pod/ServiceMonitor you can define the port name yourself.

Yes it is, since the global `ServiceMonitor` only picks up ports named `metrics`. You could theoretically have the operator define its own `ServiceMonitor`, but that locks the operator into requiring Prometheus-Operator specifically.

> It is not possible. You cannot have the same port with different names.

`ServicePort.port` must be unique, but you can use `.targetPort` to remap it to another container port (and duplicates _are_ allowed here). For example, the following works for me and is picked up correctly by Prometheus:

```rust
fn service_ports() -> Vec<ServicePort> {
    vec![
        ServicePort {
            name: Some(APP_PORT_NAME.to_string()),
            port: APP_PORT.into(),
            protocol: Some("TCP".to_string()),
            ..ServicePort::default()
        },
        ServicePort {
            name: Some("metrics".to_string()),
            port: 9504, // Arbitrary port number, this is never actually used anywhere
            protocol: Some("TCP".to_string()),
            target_port: Some(IntOrString::String(APP_PORT_NAME.to_string())),
            ..ServicePort::default()
        },
    ]
}
```